my-dsl.md

```diff
@@ -1,38 +1,96 @@
+_Fowler writes about a spectrum of languages, from general-purpose 
+languages to "purely" domain-specific. Where does the DSL you chose fall on
+this spectrum, and why?_
 
+The DSL will take text input from the user, parse it and generate
+a typeset musical score in a specificed format (such as PDF or PNG).
+
+The input text has syntax which describes the music to be typeset (including
+notes, keys, accidentals, and various annotations).  As mentioned on the
+[Text input](lilypond.org/text-input.html) page of the website "Notes are
+encoded with letters and numbers. Special commands are entered with
+backslashes."  This is notably similar to LaTeX, in that there is text which
+encodes information about how the final document should appear, which is
+then compiled to produce a beautiful document.
+
+For example, the following code, with LaTeX reminiscent syntax:
+![input-annotate.png](http://lilypond.org/pictures/text-input-1-annotate.png)
+
+produces the following output:
+
+![output.png](http://lilypond.org/pictures/text-input-1-output.png)
 
 # DSL-ness
-_Fowler writes about a spectrum of languages, from general-purpose languages to 
+_Fowler writes about a spectrum of languages, from general-purpose languages to
 "purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
 and why?_ 
 
+LilyPond is on the domain-specific side of the spectrum.  Although it can be
+used to typeset many forms of music with great generality, its use case is
+exclusively the domain of typesetting music, making it _more_ restrictive than
+TeX or LaTeX, which we already consider to be a DSL (even thought it
+is Turing Complete).
+
 
 # Internal or external?
 _Is the language implemented as an internal or external DSL? 
 Justify your answer._
 
+The language is implemented as an external DSL.  Although based on the syntax
+I initially assumed that it was an internal DSL of LaTeX, that provided
+macros or wrappers for TikZ commands, it turns out that this was incorrect.
+LilyPond is an external DSL, implemented in a mix of C++ and Lisp, and
+simply expects its input text in a format similar to TeX.
 
 # Host language
-_What language(s) was (were) used to implement the DSL?_
+_What languages were used to implement the DSL?_
 
+As mentioned in the above question, C++ and Lisp were used to implement this
+DSL.
 
 # Benefits
 _Identify one potential benefit of the DSL: how is a programmer's life or a 
 company's bottom line made easier by the existence of this language?_
 
+A programmer need not use TikZ to typeset their notes directly, nor be forced
+to make the choice between a simple framework for plotting single notes and a
+complex system that covers all the use cases of DSLs.
+
+Of course, this benefit alone could be provided simply by a library, why does
+it require its own DSL?  The major benefit of having it be its own language
+is the extra readability and ease of use, since there is no need to learn the
```

Comment:
Readability and ease of use (for domain experts) will be a very consistent theme will good, well-designed DSLs.